crates/ruff_linter/src/rules/flake8_cognitive_complexity/rules/function_is_too_cognitive_complex.rs

```diff
@@ -0,0 +1,281 @@
+            max_cognitive_complexity,
+        } = self;
+        format!("`{name}` is too cognitive complex ({complexity} > {max_cognitive_complexity})")
+    }
+}
+
+fn get_expt_cognitive_complexity(test: &Expr) -> usize {
+    let mut expt_cognitive_complexity: usize = 0;
+    if test.is_bool_op_expr() {
+        let bool_op = test.as_bool_op_expr().unwrap();
+        expt_cognitive_complexity += 1;
+        for expr in &bool_op.values {
+            expt_cognitive_complexity += get_expt_cognitive_complexity(expr);
+        }
+    }
+    expt_cognitive_complexity
+}
+
+fn get_cognitive_complexity_number(stmts: &[Stmt], nesting: usize) -> usize {
+    let mut complexity: usize = 0;
+    for stmt in stmts {
+        match stmt {
+            Stmt::If(ast::StmtIf {
+                test,
+                body,
+                elif_else_clauses,
+                ..
+            }) => {
+                complexity += 1 + nesting;
+                complexity += get_cognitive_complexity_number(body, nesting + 1);
+                for clause in elif_else_clauses {
+                    complexity += 1;
+                    complexity += get_cognitive_complexity_number(&clause.body, nesting + 1);
+                }
+                complexity += get_expt_cognitive_complexity(test);
+            }
+            Stmt::Break(..) | Stmt::Continue(..) => {
+                complexity += 1;
+            }
+            Stmt::For(ast::StmtFor { body, orelse, .. }) => {
+                complexity += 1 + nesting;
+                complexity += get_cognitive_complexity_number(body, nesting + 1);
+                complexity += get_cognitive_complexity_number(orelse, nesting + 1);
+            }
+
+            Stmt::With(ast::StmtWith { body, .. }) => {
+                complexity += get_cognitive_complexity_number(body, nesting + 1);
+            }
+            Stmt::While(ast::StmtWhile { body, orelse, .. }) => {
+                complexity += 1 + nesting;
+                complexity += get_cognitive_complexity_number(body, nesting + 1);
+                complexity += get_cognitive_complexity_number(orelse, nesting + 1);
+            }
+            Stmt::Match(ast::StmtMatch { cases, .. }) => {
+                complexity += 1 + nesting;
+                for case in cases {
+                    complexity += get_cognitive_complexity_number(&case.body, nesting + 1);
+                }
+            }
+            Stmt::FunctionDef(ast::StmtFunctionDef { body, .. }) => {
+                complexity += get_cognitive_complexity_number(body, nesting); // TODO: Needs to add nesting if nested func
+            }
+            Stmt::ClassDef(ast::StmtClassDef { body, .. }) => {
+                complexity += get_cognitive_complexity_number(body, nesting);
+            }
+            _ => {}
+        }
+    }
+    complexity
+}
+
+pub(crate) fn function_is_too_cognitive_complex(
+    stmt: &Stmt,
+    name: &str,
+    body: &[Stmt],
+    max_cognitive_complexity: usize,
```

Comment:
Nit: You can pass the entire function instead of individual parts
```suggestion
    definition: &FunctionDef,
    max_cognitive_complexity: usize,
```